projects/inverse/index.js

```diff
@@ -0,0 +1,205 @@
+/*==================================================
+  Modules
+  ==================================================*/
+
+const _ = require("underscore");
+const sdk = require("../../sdk");
+const abi = require("./abi.json");
+const BigNumber = require("bignumber.js");
+
+/*==================================================
+  TVL
+  ==================================================*/
+
+// Anchor
+const comptroller = "0x4dcf7407ae5c07f8681e1659f626e114a7667339";
+const ignore = ["0x7Fcb7DAC61eE35b3D4a51117A7c58D53f0a8a670"]; // anDOLA will be counted through the stabilizer
+const anETH = "0x697b4acAa24430F254224eB794d2a85ba1Fa1FB8";
+const wETH = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+
+// Stabilizer
+const dai = "0x6B175474E89094C44Da98b954EedeAC495271d0F";
+const dola = "0x865377367054516e17014CcdED1e7d814EDC9ce4";
+
+// Vaults
+const vaults = [
+  "0x89eC5dF87a5186A0F0fa8Cb84EdD815de6047357", // inUSDC->ETH
+  "0xc8f2E91dC9d198edEd1b2778F6f2a7fd5bBeac34", // inDAI->WBTC
+  "0x41D079ce7282d49bf4888C71B5D9E4A02c371F9B", // inDAI->YFI
+  "0x2dCdCA085af2E258654e47204e483127E0D8b277", // inDAI->ETH
+];
+
+// ask comptroller for all markets array
+async function getAllTokens(block) {
+  let tokens = (
+    await sdk.api.abi.call({
+      block,
+      target: comptroller,
+      params: [],
+      abi: abi["getAllMarkets"],
+    })
+  ).output;
+  return tokens.filter(function (token) {
+    return ignore.indexOf(token) === -1;
+  });
+}
+
+async function getAllUnderlying(block, tokens) {
+  let allUnderlying = (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(
+        tokens.filter((token) => token !== anETH),
+        (token) => ({
+          target: token,
+        })
+      ),
+      abi: abi["underlying"],
+    })
+  ).output;
+
+  allUnderlying.push({
+    input: {
+      target: anETH,
+    },
+    success: true,
+    output: wETH,
+  });
+  return allUnderlying;
+}
+
+async function getCashes(block, tokens) {
+  return (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(tokens, (token) => ({
+        target: token,
+      })),
+      abi: abi["getCash"],
+    })
+  ).output;
+}
+
+async function getTotalSupplies(block, tokens) {
+  return (
+    await sdk.api.abi.multiCall({
+      block,
+      calls: _.map(tokens, (token) => ({
+        target: token,
+      })),
+      abi: abi["totalSupply"],
+    })
+  ).output;
+}
+
+async function anchorTVL(block) {
+  const balances = {};
+
+  const tokens = await getAllTokens(block);
+  const [allUnderlying, cashes] = await Promise.all([
+    getAllUnderlying(block, tokens),
+    getCashes(block, tokens),
```

Comment:
Just a check: how do other lending platforms calculate their TVL?

This current implementation gives the number that's currently sitting in Anchor available to borrow, not what was deposited as collateral. 

The latter would be something in the order of: 
 1. Get total supply of market tokens
 2. Multiply by `exchangeRateStored` to get the number of underlying tokens
 3. Format decimals correctly

Reply:
I'm not sure I agree with the above. Getting tvl on a cash basis tells us the net balance considering borrowings that are currently out of the system. By that I mean, if something is borrowed shouldn't be counted as locked.

Reply:
Agreed with @TSnark, TVL should be amount of available liquidity, not total supplied amount